### PR TITLE
convert to typescript + apollo

### PR DESCRIPTION
**Pull request by dsifford**
Still a WIP.

**Todo:**
- [x] Drop axios from `Contact` and use a mutation to post the form to the server.
- [ ] Add a loading indicator for time between sending contact form message and receiving a response back from the other server.

Feedback welcomed and appreciated!

### CODE REVIEW

File: app/cards/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      }
+      created
+      updates
+      content
+    }
+  }
+`;
+
+const config = {
+  options: ({ match }) => ({ variables: { id: match.params.id } }),
```
Comment (jvoros) [inline app/cards/Card.tsx]: For the life of me, can't figure out where `match` variable gets set to be passed in to these config options

**Comment by dsifford** [comment]
@jvoros match is provided by react router

File: app/cards/Card.tsx (hunk 1 above)
Comment (dsifford) [inline app/cards/Card.tsx]: `match` is provided by react router.

`options` is a function which receives the `props` object at runtime and returns an object containing the variables needed to run the query. In this case, we're using `props.match`, or `{ match }` as a shorthand and setting that value of `props.match.params.id` to `id` in the graphql query.

**Comment by dsifford** [comment]
@jvoros any other thoughts on this?

**Pull request merged**